hypershift-operator/metrics_test.go

```diff
@@ -137,3 +141,100 @@ func TestMetrics(t *testing.T) {
 		})
 	}
 }
+
+func TestOperatorInfo(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		name                 string
+		hypershiftPod        *corev1.Pod
+		hypershiftDeployment *appsv1.Deployment
+		expectedImage        string
+		expectedImageID      string
+		expectedSeconds      float64
+		unexpectedSeconds    float64
+	}{
+		{
+			name: "Get hypershift operator image details from pod",
+			hypershiftPod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "operator-fakepod",
+					Namespace: "hypershift",
+				},
+				Status: corev1.PodStatus{
+					Conditions: []corev1.PodCondition{
+						{
+							Type:               corev1.PodReady,
+							Status:             corev1.ConditionTrue,
+							LastTransitionTime: metav1.Time{Time: time.Time{}.Add(10 * time.Minute)},
+						},
+					},
+					ContainerStatuses: []corev1.ContainerStatus{
+						{
+							Name:    assets.HypershiftOperatorName,
+							Image:   "quay.io/hypershift/hypershift:latest",
+							ImageID: "quay.io/hypershift/hypershift@sha256:746dc6b979cd6d42d4e6f3214e20eb5a2e0a2b4d1c1345f6914e5492e69d9afe",
+							State: corev1.ContainerState{
+								Running: &corev1.ContainerStateRunning{
+									StartedAt: metav1.Time{Time: time.Time{}},
+								},
+							},
+						},
+					},
+				},
+			},
+			hypershiftDeployment: &appsv1.Deployment{},
+			expectedImage:        "quay.io/hypershift/hypershift:latest",
+			expectedImageID:      "quay.io/hypershift/hypershift@sha256:746dc6b979cd6d42d4e6f3214e20eb5a2e0a2b4d1c1345f6914e5492e69d9afe",
+			expectedSeconds:      float64(600),
+			unexpectedSeconds:    float64(0),
+		},
+		{
+			name: "Get hypershift operator image details from deployment",
+			hypershiftPod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "operator-pod",
+					Namespace: "hypershift",
+				},
+				Status: corev1.PodStatus{},
+			},
+			hypershiftDeployment: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "operator",
+					Namespace: "hypershift",
+				},
+				Spec: appsv1.DeploymentSpec{
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Name:  "operator",
+									Image: "quay.io/hypershift/hypershift:latest",
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedImage:     "quay.io/hypershift/hypershift:latest",
+			expectedImageID:   "not found",
+			expectedSeconds:   float64(0),
+			unexpectedSeconds: float64(600),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Setenv("MY_NAMESPACE", "hypershift")
```

Comment:
seems you're using operator-fakepod as a binding name for the test outcome, so it should be declared only once in a variable.

Reply:
Still seeing operator-fakepod hardcoded

Reply:
Ohh now I got what you meant. Thanks